Compilation requires c89

If I try to build on macos 11.5, bash 3.2 and 4.4 fail because of implicit function declarations. Bash 5.1 works ok.

Based on this - https://www.gnu.org/software/bash/manual/html_node/Compilers-and-Options.html, the solution seems to be to enable c89 mode.

```
./support/bashversion.c:138:21: error: implicit declaration of function 'shell_version_string' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    printf ("%s\n", shell_version_string ());
                    ^
./support/bashversion.c:140:5: error: implicit declaration of function 'show_shell_version' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    show_shell_version (0);
    ^
```

--- a/build.sh
+++ b/build.sh
@@ -101,6 +101,8 @@ else
   if [ "$platform" = "Darwin" ]; then
     # set minimum version of macOS to 10.13
     export MACOSX_DEPLOYMENT_TARGET="10.13"
+    # https://www.gnu.org/software/bash/manual/html_node/Compilers-and-Options.html
+    export CC="gcc -std=c89 -Wno-implicit-function-declaration -Wno-return-type"
   fi
 fi
 

